Sources/Testing/Test+Macro.swift

```diff
@@ -542,7 +542,29 @@ public func __invokeXCTestCaseMethod<T>(
 @_alwaysEmitConformanceMetadata
 public protocol __TestContainer {
   /// The set of tests contained by this type.
-  static var __tests: [Test] { get async }
+  ///
+  /// - Note: This property will be removed in a future update to the testing
+  ///   library. It remains here for binary compatibility with recently-compiled
+  ///   test targets.
+  static var __tests: [Test] { get }
+
+  /// Get the set of tests contained by this type.
+  ///
+  /// - Returns: An array of ``Test`` instances.
+  ///
+  /// - Warning: This method is used to implement the `@Test` macro. Do not use
+  ///   it directly.
+  static func __tests() async -> [Test]
+}
+
+extension __TestContainer {
+  public static var __tests: [Test] {
+    []
+  }
+
+  public static func __tests() async -> [Test] {
```

Comment:
I understand the need for the default implementation of `var __tests` above, but do we need this one?

Reply:
Yes, if we have an old compiled test running on a newer build of swift-testing, we'll hit this implementation.